File: pset6/readability.py
from cs50 import get_string


def count_letters(text):
    count = 0
    for ch in text:
        if ch.isalpha() == True:
            count += 1
    return count


def main():
    text = get_string("Text: ")
    count = count_letters(text)
    words = 1
    sentence = 0
    for ch in text:
        if ch == ' ':
            words += 1
        if ch == '?' or ch == '!' or ch == '.':
            sentence += 1

    L = (count * 100.0) / words
    S = (sentence * 100.0) / words
    index = round(0.0588 * L - 0.296 * S - 15.8)
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {index}")


main()

File: pset2/readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <stdlib.h>
#include <ctype.h>
#include <math.h>

int count_letters(string text);

int main(int argc, string argv[])
{
    string text = get_string("Text: ");

    int count = count_letters(text);
    int words = 1;
    int sentence = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        char ch = text[i];
        if (ch == ' ')
        {
            words ++;
        }
        if (ch == '?' || ch == '!' || ch == '.')
        {
            sentence ++;
        }
    }

    float L = (count * 100.0) / words;
    float S = (sentence * 100.0) / words;
    int index = round(0.0588 * L - 0.296 * S - 15.8);
    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

    return 0;
}

int count_letters(string text)
{
    int count = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        char x = text[i];
        if (isalpha(x))
        {
            count ++;
        }
    }
    return count;
}


